others/iterative_tree_traversals.cpp

```diff
@@ -0,0 +1,205 @@
+ */
+Node *  BinaryTree::createNewNode(int data){
+    Node *node = new Node();
+    node->data = data;
+    node->left = node->right = nullptr;
+    return node;
+}
+
+/**
+ * @brief preOrderIterative() function that will perform the preorder traversal iteratively,
+ * and return the result array that contain the preorder traversal of a tree.
+ * @param root head/root node of the tree
+ * @return result that is containing the preorder traversal of a tree
+ */
+std::vector<int> BinaryTree::preOrderIterative(Node *root){   
+    std::stack<Node *> stack;  ///< is used to find and traverse the child nodes
+    std::vector<int> result;  ///< list of values, sorted in post-order.
+
+    
+    stack.push(root);
+    
+    while(!stack.empty()){
+        result.push_back(stack.top()->data);
+        Node *current = stack.top();
+        stack.pop();
+        
+        if(current->right){
+            stack.push(current->right);
+        }
+        if(current->left){
+            stack.push(current->left);
+        }
+    }
+    
+    return result;
+}
+
+/**
+ * @brief postOrderIterative() function that will perform the postorder traversal iteratively,
+ * and return the result array that contain the postorder traversal of a tree.
+ * @param root head/root node of a tree
+ * @return result that is containing the postorder traversal of a tree
+ */
+std::vector<int> BinaryTree::postOrderIterative(Node *root){   
+    std::stack<Node *> stack;  ///< is used to find and traverse the child nodes. 
+    std::vector<int> result;  ///< List of values, sorted in post-order.
+
+    
+    stack.push(root);
+    
+    while(!stack.empty()){
+        result.push_back(stack.top()->data);
+        Node *current = stack.top();
+        stack.pop();
+        
+        if(current->left){
+            stack.push(current->left);
+        }
+        if(current->right){
+            stack.push(current->right);
+        }
+    }
+
+    reverse(result.begin(), result.end());
+    
+    return result;
+}
+
+/**
+ * @brief inOrderIterative() function that will perform the inorder traversal iteratively,
+ * and return the result array that contain the inorder traversal of a tree.
+ * @param root head/root node of a tree
+ * @return result that is containing the inorder traversal of a tree
+ */
+std::vector<int> BinaryTree::inOrderIterative(Node *root){
+        std::stack<Node *> stack;  ///< is used to find and traverse the child nodes.
+        std::vector<int> result;  ///< List of values, sorted in in-order.
+
+    
+        Node *current = root;
+        
+        while(!stack.empty() || current){
+            while(current){
+                stack.push(current);
+                current = current->left;
+            }
+            current = stack.top();
+            stack.pop();    
+            result.push_back(current->data);   
+            current = current->right;
+        }   
+        return result;
+    }
+

```

Comment:
Please add self-test implementations using `assert` (requires the `cassert` library).